services/net/image/ImageAction.cs

```diff
@@ -77,39 +77,47 @@ public ImageAction(IApiService api, IOptions<ImageOptions> options, ILogger<Imag
+                var keyFile = new PrivateKeyFile(sshKeyFile);
+                var keyFiles = new[] { keyFile };
+                var connectionInfo = new ConnectionInfo(hostname,
+                                                    username,
+                                                    new PrivateKeyAuthenticationMethod(username, keyFiles));
+                using var client = new SftpClient(connectionInfo);
+                client.Connect();
+                remotePath = remotePath.Replace("~/", $"{client.WorkingDirectory}/");
+                var files = await FetchImagesAsync(client, remotePath);
+                files = files.Where(f => f.Name.Contains(inputFileName));
+                this.Logger.LogDebug("{count} files were discovered that match the filter '{filter}'.", files.Count(), inputFileName);
+
+                foreach (var file in files)
                 {
-                    // If another process has it in progress only attempt to do an import if it's
-                    // more than an 5 minutes old. Assumption is that it is stuck.
-                    reference = await UpdateContentReferenceAsync(reference, WorkflowStatus.InProgress);
+                    var reference = await this.FindContentReferenceAsync(manager.Ingest.Source?.Code, file.Name);
+                    if (reference == null)
+                    {
+                        reference = await this.Api.AddContentReferenceAsync(CreateContentReference(manager.Ingest, file.Name));
+                    }
+                    else if (reference.Status == (int)WorkflowStatus.InProgress && reference.UpdatedOn?.AddMinutes(5) < DateTime.UtcNow)
+                    {
+                        // If another process has it in progress only attempt to do an import if it's
+                        // more than an 5 minutes old. Assumption is that it is stuck.
+                        reference = await UpdateContentReferenceAsync(reference, WorkflowStatus.InProgress);
+                    }
+                    else continue;
+
+                    await CopyImageAsync(client, manager.Ingest, remotePath.CombineWith(file.Name));
+                    reference = await FindContentReferenceAsync(reference?.Source, reference?.Uid);
+                    if (reference != null) await ContentReceivedAsync(manager, reference, CreateSourceContent(manager.Ingest, reference));
                 }
-                else continue;
 
-                await CopyImageAsync(client, manager.Ingest, remotePath.CombineWith(file.Name));
-                reference = await FindContentReferenceAsync(reference?.Source, reference?.Uid);
-                if (reference != null) await ContentReceivedAsync(manager, reference, CreateSourceContent(manager.Ingest, reference));
+                client.Disconnect();
+            }
+            catch (Exception e)
+            {
+                Logger.LogError(e, "Failed in {class}.{method}", nameof(ImageAction), nameof(PerformActionAsync));
```

Comment:
I'm trying to understand what you're trying to obtain by adding this log here? If I'm not mistaken this exception will bubble up to the IngestManager`.cs and be logged there on line 146. Additionally, you are not adding any information that is useful to the log.

I also thought that you had determined the concurrency error was being caused by the syndication service?

Reply:
Removed, there is no need for this and for some reasons I thought I have already cleaned it up previously. Thanks